.github/workflows/pull_request.yml

```diff
@@ -50,6 +50,360 @@ jobs:
       with:
         image: "openfoodfacts-server/backend:dev"
 
+  check_perl_formatting:
+    name: Check perl formatting
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 50
+    - name: install cpanminus to install perltidy
+      run: sudo apt-get install -y cpanminus
+    - name: install perltidy
+      run: sudo cpanm Perl::Tidy
+    - name: temp list of perl files
+      run: |
+        cat << EOF > /tmp/perlfiles
```

Comment:
I would instead use an exclusion list, rather than a list, we will forgot to update it.

I think of (taken from https://stackoverflow.com/a/22558474/2886726)
```
find . -name "*.pm" -or -name "*.pl" | grep -vFf .perltidy_skip
```

`.perltidy_skip` would be a file in the gitrepo

At the end you can have a single line:
```
find  . -name "*.pm" -or -name "*.pl" | grep -vFf .perltidy_skip|xargs perltidy --assert-tidy 
```